integration/test/Test/Brig.hs

```diff
@@ -18,14 +19,45 @@ import Testlib.Prelude
 testSearchContactForExternalUsers :: HasCallStack => App ()
 testSearchContactForExternalUsers = do
   owner <- randomUser OwnDomain def {BrigI.team = True}
-  partner <- randomUser OwnDomain def {BrigI.team = True}
+  tid <- owner %. "team" & asString
 
-  bindResponse (GalleyI.putTeamMember partner (partner %. "team") (API.teamRole "partner")) $ \resp ->
+  partner <- createTeamMemberWithRole owner tid "partner"
+  tm1 <- createTeamMember owner tid
+  tm2 <- createTeamMember owner tid
+
+  -- a team member can search for contacts
+  bindResponse (BrigP.searchContacts tm1 (owner %. "name") OwnDomain) $ \resp ->
     resp.status `shouldMatchInt` 200
 
+  -- a partner is not allowed to search for contacts
   bindResponse (BrigP.searchContacts partner (owner %. "name") OwnDomain) $ \resp ->
     resp.status `shouldMatchInt` 403
 
+  -- a team member can see all other team members
+  bindResponse (Galley.getTeamMembers tm1 tid) $ \resp -> do
+    resp.status `shouldMatchInt` 200
+    assertContainsUserIds resp [owner, tm1, tm2, partner]
+
+  -- an external partner should see the person who invited them
+  bindResponse (Galley.getTeamMembers partner tid) $ \resp -> do
+    resp.status `shouldMatchInt` 200
+    assertContainsUserIds resp [owner, partner]
+
+  -- the team owner creates a conversation with the partner and another team member
+  void $ postConversation owner (defProteus {qualifiedUsers = [tm1, partner], team = Just tid}) >>= getJSON 201
+
+  -- now the external partner should also see the other team member
+  bindResponse (Galley.getTeamMembers partner tid) $ \resp -> do
+    resp.status `shouldMatchInt` 200
+    assertContainsUserIds resp [owner, tm1, partner]
+  where
+    assertContainsUserIds :: Response -> [Value] -> App ()
+    assertContainsUserIds resp users = do
+      members <- resp.json %. "members" & asList
+      userIds <- for members (\m -> m %. "user")
+      expected <- for users objId
+      userIds `shouldMatchSet` expected
```

Comment:
It looks a bit awkward that we're testing galley features in the module called `Test.Brig`. Perhaps we should just move this half of the test to the other module? 
Although I think separating these "integration" tests by where most of the code lives is not very nice. We should probably name our testing modules according to the domain.